src/zh/tongli/src/eu/kanade/tachiyomi/extension/zh/tongli/TongliDto.kt

```diff
@@ -0,0 +1,81 @@
+package eu.kanade.tachiyomi.extension.zh.tongli
+
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.JsonNames
+
+@Serializable
+class PopularResponseDto(@SerialName("RankingSet") val rankingSet: List<RankingSetDto>)
+
+@Serializable
+class RankingSetDto(@SerialName("Week") val week: List<MangaDto>)
+
+@Serializable
+class MangaDto(
+    @JsonNames("BookTitle", "Title") private val bookTitle: String,
+    @JsonNames("BookCoverURL", "CoverURL") private val bookCoverURL: String,
+    @SerialName("BookGroupID") private val bookGroupID: String,
+    @SerialName("IsSerial") private val isSerial: Boolean,
+) {
+    fun toSManga() = SManga.create().apply {
+        url = "/Book/BookVol/$bookGroupID?bookID=null&isSerial=$isSerial"
+        title = bookTitle
+        thumbnail_url = bookCoverURL
+    }
+}
+
+@Serializable
+class LatestResponseDto(
+    @SerialName("TotalPage") val totalPage: Int,
+    @SerialName("Page") val page: Int,
+    @SerialName("Books") val books: List<MangaDto>,
+)
+
+@Serializable
+class ChapterDto(
+    @SerialName("BookID") val bookID: String,
+    @SerialName("Vol") private val vol: String,
+    @SerialName("IsUpcoming") private val isUpcoming: Boolean,
+    @SerialName("IsPurchased") private val isPurchased: Boolean,
+    @SerialName("IsFree") private val isFree: Boolean,
+) {
+    fun toSChapter() = SChapter.create().apply {
+        url = if (isUpcoming) "" else "/Comic/sas/$bookID"
+        // Append lock emoji to name if not readable
```

Comment:
```suggestion
        // Prepend lock emoji to name if not readable
```

I suggest adding a space after the emoji; feel free if you don't like.